From e3dc91bce6e6ab4aa0bf3b7451bbbd71d519330e Mon Sep 17 00:00:00 2001
From: Egor Pasko <pasko@google.com>
Date: Tue, 18 May 2010 15:46:03 +0200
Subject: [PATCH 09/12] [gcc] write tableswitch relocation data to the .rodata section

by default gcc-4.4.x would put it to .text section, which does not pass the
NaCl validator.
---
 gcc/gcc/config/i386/nacl.h |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/gcc/gcc/config/i386/nacl.h b/gcc/gcc/config/i386/nacl.h
index 6898e3f..c22ad1c 100644
--- a/gcc/gcc/config/i386/nacl.h
+++ b/gcc/gcc/config/i386/nacl.h
@@ -233,6 +233,11 @@ Boston, MA 02111-1307, USA.  */
 #undef TARGET_SUBTARGET64_DEFAULT
 #define TARGET_SUBTARGET64_DEFAULT 0
 
+/* Configure script incorrectly detects this GAS capability on x86-64 and hence
+ * forces JUMP_TABLES_IN_TEXT_SECTION which cannot validate in NaCl. */
+#undef HAVE_AS_GOTOFF_IN_DATA
+#define HAVE_AS_GOTOFF_IN_DATA 1
+
 /* NaCl reserves R15 and makes RBP special in x86-64 mode */
 #undef FIXED_REGISTERS
 #define FIXED_REGISTERS						\
-- 
1.7.0.1

